ambry-store/src/main/java/com.github.ambry.store/CompactionManager.java

```diff
@@ -41,15 +38,13 @@
    * @throws StoreException when {@link BlobStore} is not started
    */
   CompactionDetails getCompactionDetails(BlobStore blobStore) throws StoreException {
-    long usedCapacity = blobStore.getSizeInBytes();
-    long totalCapacity = blobStore.getCapacityInBytes();
-    CompactionDetails details = null;
-    if (usedCapacity >= (storeConfig.storeMinUsedCapacityToTriggerCompactionInPercentage / 100.0) * totalCapacity) {
-      List<String> potentialLogSegments = blobStore.getLogSegmentsNotInJournal();
-      if (potentialLogSegments != null) {
-        details = new CompactionDetails(time.milliseconds() - messageRetentionTimeInMs, potentialLogSegments);
-      }
+    CompactionPolicyFactory compactionPolicyFactory;
+    try {
+      compactionPolicyFactory = Utils.getObj(storeConfig.storeCompactionPolicyFactory, storeConfig, time);
+    } catch (Exception e) {
+      throw new IllegalStateException("Error creating compaction policy for store " + blobStore.toString());
     }
-    return details;
+    return blobStore.getCompactionDetails(compactionPolicyFactory.getCompactionPolicy());
```

Comment:
Also, do you envision a scenario where a new `CompactionPolicy` instance has to be constructed each time?